tests/kerastuner/engine/hyperparameters_test.py

```diff
@@ -68,6 +69,101 @@ def test_name_scope():
     }
 
 
+def test_conditional_scope():
+    hp = hp_module.HyperParameters()
+    hp.Choice('choice', [1, 2, 3], default=2)
+    with hp.conditional_scope('choice', [1, 3]):
+        child1 = hp.Choice('child_choice', [4, 5, 6])
+    with hp.conditional_scope('choice', 2):
+        child2 = hp.Choice('child_choice', [7, 8, 9])
+    assert hp.values == {
+        'choice': 2,
+        'choice=1|3/child_choice': 4,
```

Comment:
Should we be concerned that the names may get very long as the list of applicable values grows?

Reply:
It could be an issue. I think for most use-cases, people will use conditionals with an hp.Choice with one or a small set of values like "model_type=DNN,Linear/learning_rate" (vizier docs even say that conditionals with multiple parent_values are generally a bad practice unless you know what you're doing) 

We could probably restructure the HyperParameters `state` and `values` to a nested structure to keep track of conditionals if we wanted to keep the conditionals out of the HP's name. 

Either way the API can stay the same, since the fact that conditionals are implemented as name_scopes is abstracted away from the user by `get`